disperser/controller/metrics/server_metrics.go

```diff
@@ -0,0 +1,112 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	grpcprom "github.com/grpc-ecosystem/go-grpc-middleware/providers/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc"
+)
+
+const namespace = "eigenda_controller"
+
+// Encapsulates metrics for the controller GRPC server
+type ServerMetrics struct {
+	logger           logging.Logger
+	grpcServerOption grpc.ServerOption
+
+	// AuthorizePayment metrics
+	authorizePaymentLatency           *prometheus.SummaryVec
+	authorizePaymentSignatureLatency  *prometheus.SummaryVec
+	authorizePaymentAuthFailures      *prometheus.CounterVec
+	authorizePaymentSignatureFailures *prometheus.CounterVec
+}
+
+func NewServerMetrics(registry *prometheus.Registry, logger logging.Logger) *ServerMetrics {
+	if registry == nil {
+		registry = prometheus.NewRegistry()
+	}
```

Comment:
```suggestion
	if registry == nil {
		return nil
	}
```

Then, make sure in each of the methods you handle the case where `m == nil`.

Creating new registries is a foot-gun pattern, since if it happens in a production environment then you may get funky results. Better to be a no-op than to register a second registry.